sdcm/nemesis.py

```diff
@@ -80,15 +81,20 @@ def update_stats(self, disrupt, status=True, data={}):
         if self.tester.create_stats:
             self.tester.update({'nemesis': self.stats})
 
-    def set_target_node(self):
+    def set_target_node(self, nemesis_start=False):
         filter_seed = self.cluster.params.get('nemesis_filter_seeds', default=True)
-        if filter_seed:
-            non_seed_nodes = [node for node in self.cluster.nodes if not node.is_seed]
-            # if non_seed_nodes is empty, nemesis would failed.
-            self.target_node = random.choice(non_seed_nodes)
-        else:
-            self.target_node = random.choice(self.cluster.nodes)
-        self.log.info('Current Target: %s', self.target_node)
+
+        with self.lock:
```

Comment:
updating `running_nemesis` logic should be on the node object, so you will not need to pass lock object here

Reply:
(put locking of shared resource logic in setter method)

Reply:
Done